docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,522 @@
+                                          // SMT in BIOS, also he/she needs to
+                                          // apply LVI mitigation in enclave
+
+Note: SGX TCB = Platform CPU HW/FW TCB + SGX PCE TCB.
+```
+If the verifier needs to provide a different quote verification policy
+beyond the policy enforced by the sgx_qv_verify_quote() API, the verifier can
+request sgx_qv_verify_quote() API to return supplemental data. The detailed
+supplemental data definition is described in above `Background` section.
+
+So we also propose to add SGX supplemental data fields to the list of claims.
+
+[PR 3353](https://github.com/openenclave/openenclave/pull/3353) added SGX
+endorsements into SGX-specific claims, here we propose to use SGX supplemental
+data (returned by QVL/QvE) instead. Because user needs to parse the endorsements
+by himself if we only return the endorsements directly, but parsing x509 cert
+chain/CRL and JSON is quite a complex task.
+
+The new SGX-Specific claims definition (Pls refer to `Background` section for
+the description of each SGX specific claim):
+
+```C
+#define OE_CLAIM_SGX_QUOTE_VERIFICATION_STATUS "sgx_quote_verification_status"
+
+#define OE_CLAIM_SGX_TCB_LEVEL_DATE_TAG "sgx_tcb_level_date_tag"
+#define OE_CLAIM_SGX_PCK_CRL_NUM "sgx_pck_crl_num"
+#define OE_CLAIM_SGX_ROOT_CA_CRL_NUM "sgx_root_ca_crl_num"
+#define OE_CLAIM_SGX_TCB_EVAL_REF_NUM "sgx_tcb_eval_ref_num"
+#define OE_CLAIM_SGX_ROOT_KEY_ID "sgx_root_key_id"
+#define OE_CLAIM_SGX_PCK_PPID "sgx_pck_ppid"
+#define OE_CLAIM_SGX_TCB_CPUSVN "sgx_tcb_cpusvn"
+#define OE_CLAIM_SGX_TCB_PCE_ISVSVN "sgx_tcb_pce_isvsvn"
+#define OE_CLAIM_SGX_PCE_ID "sgx_pce_id"
+#define OE_CLAIM_SGX_TYPE "sgx_type"
+#define OE_CLAIM_SGX_PLATFORM_INSTANCE_ID "sgx_platform_instance_id"
+#define OE_CLAIM_SGX_DYNAMIC_PLATFORM "sgx_dynamic_platform"
+#define OE_CLAIM_SGX_CACHED_KEYS "sgx_cached_keys"
+#define OE_CLAIM_SGX_SMT_ENABLED "sgx_smt_enabled"
+
+#define OE_SGX_CLAIMS_COUNT 15
+```
+
+#### Options for retrieving SGX endorsements when using QVL-based quote verification
+There are two options for the OE SDK plugin library to retrieve SGX endorsements.
+
+##### Option 1: Keep existing implementation in SGX verifier plugin
+In the current plugin implementation, if the verifier doesn't provide SGX endorsements, it
+calls API `oe_get_sgx_endorsements()` to parse SGX ECDSA quote body to get PCK
+cert chain first, then calls OCALL API `oe_get_quote_verification_collateral_ocall()`
+to load Quote Provide Library(QPL) and connect to PCK Cert Caching Sever(PCCS)
+to get corresponding verification collaterals, including CRL, TCB and QE
+Identity, etc.
+
+##### Option 2: Ask DCAP QVL to retrieve endorsements
+If the verifier doesn't provide endorsements when calling DCAP QVL API, the QVL API
+implementation will parse quote body to get PCK cert chain, then load QPL
+and connect to PCCS to get corresponding verification
+collaterals.
+
+##### Proposal: Ask DCAP QVL to retrieve endorsements
```

Comment:
If QVL/QvE code path is chosen, the endorsements are going to be impacted. So this really is a decision to be made in step 1. Right?

Reply:
The endorsements change will be in step 2. In steps, it will pass endorsements from OE to QVL directly.